iceoryx2-bb/container/src/semantic_string.rs

```diff
@@ -354,6 +354,20 @@ macro_rules! semantic_string {
             }
         }
 
+        impl From<$string_name> for String {
+            fn from(value: $string_name) -> String {
+                // SAFETY: every semantic string shall contain only valid utf-8 characters
```

Comment:
The comment is a bit confusing. What do you think of something like
```
It is ensured that the semantic string contains only valid utf-8 strings
```

On the other hand, one can create an invalid utf8 string with `unsafe`. Philosophical question, should this then check for invalid characters?

You could add the utf8 requirement to `SemanticString::new_unchecked` since this does not come from the characters the user specified in the invalid check.